src/kernels/MIOpenTensorKernelsHip.cpp

```diff
@@ -65,9 +65,9 @@ extern "C" __global__ void Op1dTensorGeneric(const MIOPEN_TYPE* __restrict__ a,
                                              const uint32_t total_work,
                                              const bool use_beta)
 {
-    const MIOPEN_TYPE* __restrict__ a_off = a + Aoffset;
-    const MIOPEN_TYPE* __restrict__ b_off = b + Boffset;
-    MIOPEN_TYPE* __restrict__ c_off       = c + Coffset;
+    const MIOPEN_TYPE* a_off = a + Aoffset;
```

Comment:
Why __restrict__ qualifier is removed? This allows the compiler to make optimizations (for example, vectorization) that would not otherwise have been possible.

Reply:
I'm aware about it.
I removed `__restrict__` because it assumes that pointers do not overlap, but as far as I can see - we can't guarantee that, so the optimizations may not be applicable.

Reply:
Ok. It's fine to remove restrict qualifier to pointer if we can't guarantee no other pointer will be used to access the memory to which it points.